### ISSUE TITLE
Server returned status 401 (SickGear)

### ISSUE THREAD

**Issue by whvanderpost** [clinton-hall/nzbToMedia#1346]
Hi,

Since a couple of days all of a sudden I started getting errors (Server returned status 401!) in my post processing script. 401 is unauthorized right? 

- I'm running SABnzb v2.3.3 and SickGear (latest, 0.16.0) on a Synology 210+.
- I haven't changed anything in the config of the packages.
- I've pulled the latest version of the nzbToMedia scripts just to be sure.
- In my autoProcessMedia.cfg I have SickGear configured with a username/password.

Here is the log output from the script:
```
[23:22:51] [INFO]::MAIN: Loading config from [/volume1/Documents/nzbToMedia/autoProcessMedia.cfg]
[23:22:51] [INFO]::MAIN: Checking database structure...
[23:22:51] [DEBUG]::MAIN: Checking Initial Schema database upgrade
[23:22:51] [DEBUG]::MAIN: InitialSchema upgrade not required
[23:22:51] [DEBUG]::MAIN: Checking if we can use git commands: "/usr/local/git/bin/git" version
[23:22:51] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" version with your shell in /volume1/Documents/nzbToMedia
[23:22:51] [DEBUG]::MAIN: "/usr/local/git/bin/git" version : returned successful
[23:22:51] [DEBUG]::MAIN: Using: "/usr/local/git/bin/git"
[23:22:51] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" symbolic-ref -q HEAD with your shell in /volume1/Documents/nzbToMedia
[23:22:51] [DEBUG]::MAIN: "/usr/local/git/bin/git" symbolic-ref -q HEAD : returned successful
[23:22:51] [INFO]::MAIN: Checking if git needs an update
[23:22:51] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" rev-parse HEAD with your shell in /volume1/Documents/nzbToMedia
[23:22:51] [DEBUG]::MAIN: "/usr/local/git/bin/git" rev-parse HEAD : returned successful
[23:22:51] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" fetch origin with your shell in /volume1/Documents/nzbToMedia
[23:22:52] [DEBUG]::MAIN: "/usr/local/git/bin/git" fetch origin : returned successful
[23:22:52] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" rev-parse --verify --quiet "@{upstream}" with your shell in /volume1/Documents/nzbToMedia
[23:22:52] [DEBUG]::MAIN: "/usr/local/git/bin/git" rev-parse --verify --quiet "@{upstream}" : returned successful
[23:22:52] [DEBUG]::MAIN: Executing "/usr/local/git/bin/git" rev-list --left-right "@{upstream}"...HEAD with your shell in /volume1/Documents/nzbToMedia
[23:22:52] [DEBUG]::MAIN: "/usr/local/git/bin/git" rev-list --left-right "@{upstream}"...HEAD : returned successful
[23:22:52] [DEBUG]::MAIN: cur_commit = ee243fbba51e66ed12ad17d57e016b4db5bdc720 % (newest_commit)= ee243fbba51e66ed12ad17d57e016b4db5bdc720, num_commits_behind = 0, num_commits_ahead = 0
[23:22:52] [INFO]::MAIN: No update needed
[23:22:52] [INFO]::MAIN: nzbToMedia Version:ee243fbba51e66ed12ad17d57e016b4db5bdc720 Branch:master (Linux 2.6.32.12)
[23:22:52] [INFO]::MAIN: #########################################################
[23:22:52] [INFO]::MAIN: ## ..::[nzbToMedia.pyc]::.. ##
[23:22:52] [INFO]::MAIN: #########################################################
[23:22:52] [DEBUG]::MAIN: Options passed into nzbToMedia: ['/volume1/Documents/nzbToMedia/nzbToSickBeard.py', '/volume1/Download/SickGear/[some show]', '[some show].nzb', '[some show]', '', 'sickgear', 'alt.binaries.font', '0', 'https://provider/fail/[snip]/[snip]']
[23:22:52] [INFO]::MAIN: Script triggered from SABnzbd 0.7.17+
[23:22:52] [DEBUG]::MAIN: Searching for nzoid from SAbnzbd ...
[23:22:53] [ERROR]::MAIN: Unable to open URL
[23:22:53] [DEBUG]::MAIN: Adding NZB download info for directory /volume1/Download/SickGear/[some show] to database
[23:22:53] [INFO]::MAIN: Auto-detected SECTION:SickBeard
[23:22:53] [DEBUG]::MAIN: Checking for archives to extract in directory: /volume1/Download/SickGear/[some show]
[23:22:53] [INFO]::MAIN: Calling SickBeard:sickgear to post-process:[some show].nzb
[23:22:53] [DEBUG]::SERVER: Attempting to connect to server at http://127.0.0.1:8083
[23:22:53] [DEBUG]::SERVER: Server responded at http://127.0.0.1:8083
[23:22:53] [INFO]::MAIN: SickBeard:sickgear fork set to failed
[23:22:53] [INFO]::MAIN: FLATTEN: Flattening directory: /volume1/Download/SickGear/[some show]
[23:22:53] [DEBUG]::MAIN: Checking for empty folders in:/volume1/Download/SickGear/[some show]
[23:22:53] [POSTPROCESS]::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
[23:22:53] [DEBUG]::SICKBEARD: Opening URL: http://127.0.0.1:8083/home/postprocess/processEpisode with params: {'nzbName': '[some show].nzb', 'failed': '0', 'dirName': '/volume1/Download/SickGear/[some show]', 'quiet': 1}
[23:22:53] [ERROR]::SICKBEARD: Server returned status 401
[23:22:53] [ERROR]::MAIN: A problem was reported in the /volume1/Documents/nzbToMedia/nzbToSickBeard.py script.
SickBeard: Failed to post-process - Server returned status 401!
```

Can you tell me if I need to change some config somewhere?

**Comment by nikagl** [comment]
Yep, same problem here. I guess some SickGear update?

**Comment by nikagl** [comment]
Some API updates here: https://github.com/SickGear/SickGear/commit/a6f7ab47e2b8a8e5892bffea9b206e2464cdbc30

**Comment by nikagl** [comment]
I reverted to an older release (after disabling auto-updates) using the following command:
```git checkout tags/release_0.15.14```

I restarted and all is ok now. Hope anyone can see what is wrong in the code...

**Comment by clinton-hall** [comment]
What happens when you navigate to http://127.0.0.1:8083/home/postprocess
on the newer version of sickgear?

**Comment by nikagl** [comment]
Hi Clinton,

 

I have compared the two sources of the login pages and the difference is as follows:



 

So the following hidden input field was added:

<input type="hidden" name="_xsrf" value="2|4f46288f|0a2de9a6142fee4a980abf7a1919e613|1525345780">

 

For more details, I see they added an FAQ bit here:

https://github.com/SickGear/SickGear/wiki/Frequently-Answered-Questions#userpass

 

Any thoughts about their comment here?

p.s. As a courtesy, I looked to inform the author of nzbToMedia, but given SickGear isn't listed on its homepage, it is read that it doesn't fully support SickGear, therefore, may as well replace it with the official script as shown in the wiki install pages under  <https://github.com/SickGear/SickGear/wiki/Installation-Instructions#misc> Misc for NZBGet, SABnzbd or torrent clients.

Regards,

 

Nika.

**Comment by Th3MadHatter** [comment]
I have the same issue since friday, I didn't notice until now so have to do a lot manually.

nzbToMedia Version:ee243fbba51e66ed12ad17d57e016b4db5bdc720 Branch:master (Windows 10)
```
INFO    ::MAIN: Calling SickBeard:tv to post-process:removed
INFO    ::MAIN: Attempting to auto-detect tv fork
INFO    ::MAIN: SickBeard:tv fork auto-detection failed
INFO    ::MAIN: SickBeard:tv fork set to default
INFO    ::MAIN: FLATTEN: Flattening directory: removed
POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
ERROR   ::SICKBEARD: Server returned status 401
ERROR   ::MAIN: A problem was reported in the autoProcess* script. If torrent was paused we will resume seeding
ERROR   ::MAIN: A problem was reported in the C:\nzbToMedia\TorrentToMedia.py script.
```

SG returns

`403 POST /login (127.0.0.1): '_xsrf' argument missing from POST`

**Comment by whvanderpost** [comment]
@clinton-hall

> What happens when you navigate to http://127.0.0.1:8083/home/postprocess on the newer version of sickgear?

First I get the login page, after logging in I end up on the "Process Media" page.

**Comment by Th3MadHatter** [comment]
@clinton-hall Any idea if this is still fixable for SG or is it time to migrate?
I tried asking the guy at SG to add a linking option to his batch script (which only copies) but he doesn't feel like it and suggests everyone to just buy extra HDD's and do it his way...

I'm kind of stuck with this, any help/suggestion would be much appreciated :smiley:

**Comment by clinton-hall** [comment]
Hi All, Sorry for this. I hadn't had a chance to look into this any closer.

I'll make 2 changes here:

1. Add SickGear to my wiki and front page descriptions so as to "officially" support it.
2. Add user/pass and `_xsrf` as per the official autoProcessTV. 

I am not sure if I will be able to use auto-fork testing. I might need to make SickGear an option that needs to be configured in the script... I'll try and have this done sometime this week.

**Comment by clinton-hall** [comment]
Ok, after the 2 updates above (second simply because I was trying to rush this while working remotely!) this is ready to test...
In theory this should still work with auto-fork... please test in latest nightly and let me know.

If it does, I'll just need to update the wiki to show official SickGear support m(unless anyone wants to update the wiki for me ;)

**Comment by JackDandy** [comment]
Hi @clinton-hall 

Perhaps.. `timeour` should be `timout` at https://github.com/clinton-hall/nzbToMedia/commit/edd31b3eee309389faa2bd8536b46be028f648fb#diff-2cf211dd109819c2cf48f7a754048c28R273 ?

:+1:

**Comment by JackDandy** [comment]
Hi @clinton-hall Perhaps.. `timeour` should be `timeout` at [this line](https://github.com/clinton-hall/nzbToMedia/commit/edd31b3eee309389faa2bd8536b46be028f648fb#diff-2cf211dd109819c2cf48f7a754048c28R273) ? :+1:

**Comment by clinton-hall** [comment]
@JackDandy already fixed https://github.com/clinton-hall/nzbToMedia/commit/189dde26c8551e1c71fc18b9124c7d81de535ddd

**Comment by whvanderpost** [comment]
Okay so I've tested with the nightly build, apparently something still isn't working as it should.
As you can see in the copy of the Sabnzbd log below, the whole 'post-processing' page is being returned.

```
[17:10:06] [INFO]::MAIN: nzbToMedia Version:0259b83afadc365a6fb821a01d06516ecbe9a52d Branch:nightly (Linux 2.6.32.12)
[17:10:06] [INFO]::MAIN: #########################################################
[17:10:06] [INFO]::MAIN: ## ..::[nzbToMedia.pyc]::.. ##
[17:10:06] [INFO]::MAIN: #########################################################
[17:10:06] [DEBUG]::MAIN: Options passed into nzbToMedia: ['/volume1/Documents/nzbToMedia/nzbToSickBeard.py', '/volume1/Download/SickGear/[some show], '[some show].nzb', '[some show]', '', 'sickgear', 'alt.binaries.frogs', '0', 'https://nzb provider url']
[17:10:06] [INFO]::MAIN: Script triggered from SABnzbd 0.7.17+
[17:10:06] [DEBUG]::MAIN: Searching for nzoid from SAbnzbd ...
[17:10:06] [ERROR]::MAIN: Unable to open URL
[17:10:06] [DEBUG]::MAIN: Adding NZB download info for directory /volume1/Download/SickGear/[some show] to database
[17:10:06] [INFO]::MAIN: Auto-detected SECTION:SickBeard
[17:10:06] [DEBUG]::MAIN: Checking for archives to extract in directory: /volume1/Download/SickGear/[some show]
[17:10:06] [INFO]::MAIN: Calling SickBeard:sickgear to post-process:[some show].nzb
[17:10:06] [DEBUG]::SERVER: Attempting to connect to server at http://127.0.0.1
[17:10:07] [DEBUG]::SERVER: Server responded at http://127.0.0.1
[17:10:07] [INFO]::MAIN: SickBeard:sickgear fork set to failed
[17:10:07] [INFO]::MAIN: FLATTEN: Flattening directory: /volume1/Download/SickGear/[some show]
[17:10:07] [DEBUG]::MAIN: Checking for empty folders in:/volume1/Download/SickGear/[some show]
[17:10:07] [POSTPROCESS]::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
[17:10:07] [DEBUG]::SICKBEARD: Opening URL: http://127.0.0.1/home/postprocess/processEpisode with params: {'nzbName': '[some show.nzb', 'failed': '0', 'dirName': '/volume1/Download/SickGear/[some show]', 'quiet': 1}
[17:10:08] [POSTPROCESS]::SICKBEARD: <!DOCTYPE html>
[17:10:08] [POSTPROCESS]::SICKBEARD: <html>
[17:10:08] [POSTPROCESS]::SICKBEARD: <head>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<meta charset="utf-8">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<meta name="robots" content="noindex, nofollow, noarchive, nocache, noodp, noydir, noimageindex, nosnippet">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<meta name="viewport" content="width=device-width, initial-scale=1">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<meta http-equiv="X-UA-Compatible" content="IE=edge">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<title>SickGear - Process Media</title>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<!--[if lt IE 9]>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<![endif]-->
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<link rel="shortcut icon" href="/images/ico/favicon.ico">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<link rel="apple-touch-icon" sizes="180x180" href="/images/ico/apple-touch-icon-180x180.png">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<link rel="apple-touch-icon" sizes="152x152" href="/images/ico/apple-touch-icon-152x152.png">
**[snipped some html]**
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<link rel="icon" type="image/png" href="/images/ico/favicon-192x192.png" sizes="192x192">
**[snipped some html]**
[17:10:08] [POSTPROCESS]::SICKBEARD: </head>
[17:10:08] [POSTPROCESS]::SICKBEARD: <body class="">
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<h1 class="header">Process Media</h1>
[17:10:08] [POSTPROCESS]::SICKBEARD: 	<form name="processForm" method="post" action="processEpisode">
[17:10:08] [POSTPROCESS]::SICKBEARD: 		<input type="hidden" id="type" name="type" value="manual">
**[snipped some more]**
[17:10:09] [POSTPROCESS]::SICKBEARD: </script>
[17:10:09] [POSTPROCESS]::SICKBEARD: 	</footer>
[17:10:09] [POSTPROCESS]::SICKBEARD: </body>
[17:10:09] [POSTPROCESS]::SICKBEARD: </html>
[17:10:09] [ERROR]::MAIN: A problem was reported in the /volume1/Documents/nzbToMedia/nzbToSickBeard.py script.
SickBeard: Failed to post-process - Returned log from SickBeard was not as expected.!
```

**Comment by JackDandy** [comment]
Hi @whvanderpost 

I am not an expert on this, but just glancing over your logs, I see that you're not using the latest version of nzbToMedia (the tell is the case of `sickgear` is now `SickGear`), Clinton posted a fix 4 days ago so perhaps that may be your issue here ?

JD

**Comment by whvanderpost** [comment]
But the first line in the log is:
`[17:10:06] [INFO]::MAIN: nzbToMedia Version:0259b83afadc365a6fb821a01d06516ecbe9a52d Branch:nightly (Linux 2.6.32.12)`

And the latest commit (in which that casing was fixed) is:
`https://github.com/clinton-hall/nzbToMedia/commit/0259b83afadc365a6fb821a01d06516ecbe9a52d`

Really have no idea why that casing is still wrong then :-/
On my nas, the timestamp of the file nzbToSickBeard.py is also from today..

**Comment by JackDandy** [comment]
Hmm, that is strange with the output...

> [17:10:07] [INFO]::MAIN: SickBeard:sickgear fork set to failed

But like you say, your commit# are good. 

@clinton-hall will no doubt track the issues down :)

**Comment by clinton-hall** [comment]
In this case `SickBeard:sickgear` comes about because `SickBeard` is the Section (used to load the settings from the Category) and `sickrage` is the category passed into the script by the SABnzbd

```
Options passed into nzbToMedia: ['/volume1/Documents/nzbToMedia/nzbToSickBeard.py', '/volume1/Download/SickGear/[some show], '[some show].nzb', '[some show]', '', 'sickgear', 'alt.binaries.frogs', '0', 'https://nzb provider url']
```

I am not sure why this is detecting the fork as failed... what happens if you set
`fork = SickGear` instead of `fork = auto`?

For sake of Case preservation, you can change the category in SABnzbd to `SickGear` and then change this in autoProcessMedia.cfg

```
[SickBeard]
    [SickGear]
        fork = SickGear
```

**Comment by whvanderpost** [comment]
The fork property in the cfg is set to “failed”. It was one of the options that indicated using a sickbeard fork that supports failed downloads. So that’s why it’s there in the log, it’s not an error.

I’ll change the cfg (and sabnzbd settings) tomorrow...

**Comment by JackDandy** [comment]
Aaaand @clinton-hall fails to spell 8 letters of SickGear :p

JD runs and hides

**Comment by clinton-hall** [comment]
@whvanderpost 
I see where this comes from. `failed` is basically a "generic" SickBeard fork that supports failed handling, which was introduced before the later forks were added. However where there are defined branches e.g. SickGear, SickRage, Medusa etc, those should be specified...

or best option is to use `fork = auto` Please test this option first, as this will basically determine which available options exist on the processing page and then pass those parameters accordingly.

@JackDandy Fair call... I have corrected now.

**Comment by JackDandy** [comment]
> Fair call... I have corrected now.

@clinton-hall lol, you nearly got it right, same place, give your fingers or keyboard a good talking to :p

**Comment by clinton-hall** [comment]
@JackDandy 
OMG... so that is embarrassing. I am the one who should run and hide...
To be fair, I am in New Zealand, UTC + 1200... so this was an early morning post before starting work. I had not yet had my coffee at the time of attempting to correct that one.

**Comment by whvanderpost** [comment]
I've changed the fork to "auto" and now the script succeeds 👍 
Also tried to change the category in Sab, but in the UI it automatically converts the value to lowercase when the textbox loses focus. So I've left that in lowercase for now (and of course the section in the .cfg file as well)

**Comment by clinton-hall** [comment]
interesting... so SABnzbd only passes the category as lower case?... so you will see `SickBeard:sickgear fork set to SickGear`. 
If this annoys you (the changed case in SickGear), you could set Category to `tv` or `series` etc...
`SickBeard:tv fork set to SickGear`
or
`SickBeard:series fork set to SickGear`

Personally I have always set the category to be tv, movies, or music etc, to reflect the media type and not the media management programme.
Anyways, glad it is working for you. If you don't like the case display of `sickgear` in the logs, there are options. Otherwise enjoy and let me know if you find any other issues.

**Comment by whvanderpost** [comment]
Yes, you get:
```
[09:33:45] [INFO]::MAIN: Attempting to auto-detect sickgear fork
[09:33:47] [INFO]::MAIN: SickBeard:sickgear fork auto-detection successful ...
[09:33:47] [INFO]::MAIN: SickBeard:sickgear fork set to SickGear
```
It doesn't annoy me, after all .. the official sickgear logo (@JackDandy profile icon) has 'gear' in lowercase also :p

**Comment by JackDandy** [comment]
> the official sickgear logo (@JackDandy profile icon) has 'gear' in lowercase 

@whvanderpost the logo is SICKgear which although looks ok in the logo, its not so nice in ASCII, which is SickGear in all other places (see repo, wiki etc.) Anyway, no worries, I don't have to look at it *wink*

**Comment by clinton-hall** [comment]
@JackDandy 
Obviously I have tried to maintain support for a variety of Sick* repositories along with Sonarr and other applications. I have tried to remain "neutral" in discussions of preferred forks etc. I also created packages for Asustor and QNAP and ensured that these allowed installation of SickBeard, SickRage, SickRageTV, and "Custom" URL so in theory SickGear can be installed within those packages by entering the SickGear URL...
I'll see if I can update these to specifically provide a SickGear package.

One contributor to this project also maintains/maintained a Sick* fork and I believe there was a DMCA take-down issue that impacted multiple Sick* forks. Personally I was disappointed to see such action within the Open-source community.

I have never tried to push my script Vs any other, but just tried to provide some additional features that were requested etc... and I'm always happy for any other repository to take on these features, and potentially write them properly ;) , otherwise for those who want them, I try and keep them compatible.

All that said, I knew of SickGear but had not realized the level of development that has gone on... just had a bit of a read... I'll try some test cases on my QNAP and Asustor to see what I can do to package these (if not already done).

Publicly here on this project I'll never say what Media Management tools I personally run... but I'll try and keep all major versions tested and running as far as my scripts go.

**Comment by JRShield** [comment]
It took me way to long to figure out I should switch to the nightly branch :)

I can however confirm the issue is solved for me!

**Comment by nikagl** [comment]
For me too, thanks guys!

**Comment by Th3MadHatter** [comment]
Looks like the error is back over here.
Noticed a bunch of 401 messages from the last couple of days since it upgraded to .

```
2018-06-29 08:03:41 INFO    ::MAIN: nzbToMedia Version:5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 Branch:nightly (Windows 10)
2018-06-29 08:03:41 WARNING ::MAIN: Failed to locate ffmpeg.exe. Transcoding disabled!
2018-06-29 08:03:41 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2018-06-29 08:03:41 INFO    ::MAIN: #########################################################
2018-06-29 08:03:41 INFO    ::MAIN: ## ..::[TorrentToMedia.py]::.. ##
2018-06-29 08:03:41 INFO    ::MAIN: #########################################################
2018-06-29 08:03:41 INFO    ::MAIN: SEARCH: Found torrent file Test.Video.S01E02.720p.WEB.x264-TEST.mkv in input directory directory X:\Download\tv
2018-06-29 08:03:41 INFO    ::MAIN: SEARCH: Setting inputDirectory to T:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 08:03:41 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2018-06-29 08:03:46 INFO    ::MAIN: Output directory set to: X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 08:03:46 INFO    ::COPYLINK: MEDIAFILE: [Test.Video.S01E02.720p.WEB.x264-TEST.mkv]
2018-06-29 08:03:46 INFO    ::COPYLINK: SOURCE FOLDER: [X:\Download\tv]
2018-06-29 08:03:46 INFO    ::COPYLINK: TARGET FOLDER: [X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv]
2018-06-29 08:03:46 INFO    ::COPYLINK: Hard linking SOURCE MEDIAFILE -> TARGET FOLDER
2018-06-29 08:03:46 INFO    ::MAIN: FLATTEN: Flattening directory: X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 08:03:46 INFO    ::MAIN: Found 1 media files in X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 08:03:46 INFO    ::MAIN: Calling SickBeard:tv to post-process:Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 08:03:46 INFO    ::MAIN: SickBeard:tv fork set to SickGear
2018-06-29 08:03:46 POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
2018-06-29 08:03:46 ERROR   ::SICKBEARD: Server returned status 401
2018-06-29 08:03:46 ERROR   ::MAIN: A problem was reported in the autoProcess* script. If torrent was paused we will resume seeding
2018-06-29 08:03:51 ERROR   ::MAIN: A problem was reported in the C:\nzbToMedia\TorrentToMedia.py script.
```
Nothing shows up in the SickGear log.

It worked great on 0259b83afadc365a6fb821a01d06516ecbe9a52d but after the first upgrade to 33ac485e1b45666aefeb1b47b7fbd99e938d82d3 it stopped working again.
I tried fork = auto and fork = Sickgear but both give a 401, there have been no successful connections after the upgrade to 33ac485e1b45666aefeb1b47b7fbd99e938d82d3 Branch:nightly (Windows 10).

**Comment by nikagl** [comment]
With nzbToMedia I am on:
```
2018-06-29 09:32:08 DEBUG   ::MAIN: cur_commit = 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 % (newest_commit)= 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4, num_commits_behind = 0, num_commits_ahead = 0
2018-06-29 09:32:08 INFO    ::MAIN: No update needed
2018-06-29 09:32:08 INFO    ::MAIN: nzbToMedia Version:5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 Branch:nightly (Windows 7)
```

With SickGrear I am on:
```
2018-06-29 05:55:54 DEBUG    CHECKVERSION :: cur_commit = 70f1cc58ffe255a3fffedcae67337380005ac75a, newest_commit = 54a877d3e39ad7b33d7bebedfa5b91c4b2e05242, num_commits_behind = 7, num_commits_ahead = 0
```
And all is ok... 

Upgrading SickGear now to see whether the latest commit in SickGear killed something again? :-)

**Comment by clinton-hall** [comment]
@Th3MadHatter any chance you can enable debug logs?

What happens if you roll back to 0259b83 ?
`git checkout 0259b83afadc365a6fb821a01d06516ecbe9a52d .`
be sure to disable auto-update in autoProcessMedia.cfg
```
[General]
    auto_update = 0
```
if that works, then the issue was a recent update at my end.

**Comment by nikagl** [comment]
Upgraded sickgear to latest (54a877d3e39ad7b33d7bebedfa5b91c4b2e05242) and all is still fine...

nzbToMedia

```
2018-06-29 10:53:00 DEBUG   ::MAIN: cur_commit = 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 % (newest_commit)= 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4, num_commits_behind = 0, num_commits_ahead = 0
```

Sickgear

```
2018-06-29 09:46:55 DEBUG    CHECKVERSION :: cur_commit = 54a877d3e39ad7b33d7bebedfa5b91c4b2e05242, newest_commit = 54a877d3e39ad7b33d7bebedfa5b91c4b2e05242, num_commits_behind = 0, num_commits_ahead = 0
```

Here's the log snippet to show the processing:

```
2018-06-29 10:54:53 INFO    ::MAIN: SickBeard:tv fork set to SickGear
2018-06-29 10:54:55 INFO    ::SUBTITLES: Attempting to download subtitles for XXXXX
2018-06-29 10:55:07 POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
2018-06-29 10:55:07 DEBUG   ::SICKBEARD: Opening URL: http://localhost:8083/home/postprocess/processEpisode with params: {'nzbName': 'XXXXX', 'quiet': 1, 'failed': 0, 'process_method': 'copy', 'dir': 'XXXXX'}
2018-06-29 10:58:42 POSTPROCESS::SICKBEARD: Checking folder... XXXXX
etc.

```

**Comment by Th3MadHatter** [comment]
With all debugs checked
```
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: TMP: C:\Users\Computer\AppData\Local\Temp
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: COMPUTERNAME: COMPUTER-A7777
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: USERDOMAIN: COMPUTER-A7777
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PSMODULEPATH: C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: COMMONPROGRAMFILES: C:\Program Files (x86)\Common Files
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROCESSOR_IDENTIFIER: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROGRAMFILES: C:\Program Files (x86)
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROCESSOR_REVISION: 2a07
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: SYSTEMROOT: C:\Windows
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PATH: C:\Python27\;C:\Python27\Scripts;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Git\cmd;C:\Users\Computer\AppData\Local\Microsoft\WindowsApps;
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROGRAMFILES(X86): C:\Program Files (x86)
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: COMSPEC: C:\Windows\system32\cmd.exe
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: TEMP: C:\Users\Computer\AppData\Local\Temp
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: COMMONPROGRAMFILES(X86): C:\Program Files (x86)\Common Files
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROCESSOR_ARCHITECTURE: x86
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: ALLUSERSPROFILE: C:\ProgramData
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: LOCALAPPDATA: C:\Users\Computer\AppData\Local
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: FPS_BROWSER_USER_PROFILE_STRING: Default
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: HOMEPATH: \Users\Computer
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: USERDOMAIN_ROAMINGPROFILE: COMPUTER-A7777
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROGRAMW6432: C:\Program Files
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: USERNAME: Computer
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: LOGONSERVER: \\COMPUTER-A7777
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: SESSIONNAME: Console
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROGRAMDATA: C:\ProgramData
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: ONEDRIVE: C:\Users\Computer\OneDrive
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: FPS_BROWSER_APP_PROFILE_STRING: Internet Explorer
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PATHEXT: .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: WINDIR: C:\Windows
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: APPDATA: C:\Users\Computer\AppData\Roaming
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: HOMEDRIVE: C:
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: SYSTEMDRIVE: C:
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: NUMBER_OF_PROCESSORS: 8
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROCESSOR_LEVEL: 6
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PROCESSOR_ARCHITEW6432: AMD64
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: COMMONPROGRAMW6432: C:\Program Files\Common Files
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: OS: Windows_NT
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: PUBLIC: C:\Users\Public
2018-06-29 12:59:04 INFO    ::ENVIRONMENT: USERPROFILE: C:\Users\Computer
2018-06-29 12:59:04 INFO    ::MAIN: Checking database structure...
2018-06-29 12:59:04 DEBUG   ::MAIN: Checking Initial Schema database upgrade
2018-06-29 12:59:04 DB      ::MAIN: nzbtomedia.db: SELECT 1 FROM sqlite_master WHERE name = ?; with args ('db_version',)
2018-06-29 12:59:04 DB      ::MAIN: nzbtomedia.db: SELECT db_version FROM db_version
2018-06-29 12:59:04 DEBUG   ::MAIN: InitialSchema upgrade not required
2018-06-29 12:59:04 DEBUG   ::MAIN: Checking if we can use git commands: "C:\Program Files\Git\bin\git.exe" version
2018-06-29 12:59:04 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" version with your shell in C:\nzbToMedia
2018-06-29 12:59:04 DEBUG   ::MAIN: git output: git version 2.16.1.windows.2
2018-06-29 12:59:04 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" version : returned successful
2018-06-29 12:59:04 DEBUG   ::MAIN: Using: "C:\Program Files\Git\bin\git.exe"
2018-06-29 12:59:04 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" symbolic-ref -q HEAD with your shell in C:\nzbToMedia
2018-06-29 12:59:05 DEBUG   ::MAIN: git output: refs/heads/nightly
2018-06-29 12:59:05 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" symbolic-ref -q HEAD : returned successful
2018-06-29 12:59:05 INFO    ::MAIN: Checking if git needs an update
2018-06-29 12:59:05 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" rev-parse HEAD with your shell in C:\nzbToMedia
2018-06-29 12:59:05 DEBUG   ::MAIN: git output: 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4
2018-06-29 12:59:05 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" rev-parse HEAD : returned successful
2018-06-29 12:59:05 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" fetch origin with your shell in C:\nzbToMedia
2018-06-29 12:59:07 DEBUG   ::MAIN: git output: 
2018-06-29 12:59:07 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" fetch origin : returned successful
2018-06-29 12:59:07 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" rev-parse --verify --quiet "@{upstream}" with your shell in C:\nzbToMedia
2018-06-29 12:59:07 DEBUG   ::MAIN: git output: 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4
2018-06-29 12:59:07 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" rev-parse --verify --quiet "@{upstream}" : returned successful
2018-06-29 12:59:07 DEBUG   ::MAIN: Executing "C:\Program Files\Git\bin\git.exe" rev-list --left-right "@{upstream}"...HEAD with your shell in C:\nzbToMedia
2018-06-29 12:59:07 DEBUG   ::MAIN: git output: 
2018-06-29 12:59:07 DEBUG   ::MAIN: "C:\Program Files\Git\bin\git.exe" rev-list --left-right "@{upstream}"...HEAD : returned successful
2018-06-29 12:59:07 DEBUG   ::MAIN: cur_commit = 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 % (newest_commit)= 5d5d69ecdd5fe043be8a915ffa26e8427c9641c4, num_commits_behind = 0, num_commits_ahead = 0
2018-06-29 12:59:07 INFO    ::MAIN: No update needed
2018-06-29 12:59:07 INFO    ::MAIN: nzbToMedia Version:5d5d69ecdd5fe043be8a915ffa26e8427c9641c4 Branch:nightly (Windows 10)
2018-06-29 12:59:07 WARNING ::MAIN: Failed to locate ffmpeg.exe. TranscComputerg disabled!
2018-06-29 12:59:07 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2018-06-29 12:59:07 DEBUG   ::MAIN: Connecting to utorrent: http://localhost:40010/gui/
2018-06-29 12:59:08 INFO    ::MAIN: #########################################################
2018-06-29 12:59:08 INFO    ::MAIN: ## ..::[TorrentToMedia.py]::.. ##
2018-06-29 12:59:08 INFO    ::MAIN: #########################################################
2018-06-29 12:59:08 DEBUG   ::MAIN: Options passed into TorrentToMedia: ['C:\\nzbToMedia\\TorrentToMedia.py', 'X:\\Download\\tv', 'Test.Video.S01E02.720p.WEB.x264-TEST.mkv', 'tv', 'ACD1223BAAAC0344456A']
2018-06-29 12:59:08 DEBUG   ::MAIN: Adding TORRENT download info for directory X:\Download\tv to database
2018-06-29 12:59:08 DB      ::MAIN: nzbtomedia.db: UPDATE downloads SET status = ?, client_agent = ?, last_update = ?, input_id = ?, input_hash = ?, input_name = ? WHERE input_directory = ? with args [0, u'utorrent', 736874, u'158B313CAFA3333CDCB22AA', u'ACD1223BAAAC0344456A', u'Test.Video.S01E02.720p.WEB.x264-TEST.mkv', u'X:\\Download\\tv']
2018-06-29 12:59:08 DEBUG   ::MAIN: Received Directory: X:\Download\tv | Name: Test.Video.S01E02.720p.WEB.x264-TEST.mkv | Category: tv
2018-06-29 12:59:08 DEBUG   ::MAIN: SEARCH: Found the Category: tv in directory structure
2018-06-29 12:59:08 INFO    ::MAIN: SEARCH: Found torrent file Test.Video.S01E02.720p.WEB.x264-TEST.mkv in input directory directory X:\Download\tv
2018-06-29 12:59:08 INFO    ::MAIN: SEARCH: Setting inputDirectory to X:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:08 DEBUG   ::MAIN: Determined Directory: X:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv | Name: Test.Video.S01E02.720p.WEB.x264-TEST.mkv | Category: tv
2018-06-29 12:59:08 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2018-06-29 12:59:08 DEBUG   ::MAIN: Stopping torrent Test.Video.S01E02.720p.WEB.x264-TEST.mkv in utorrent while processing
2018-06-29 12:59:13 INFO    ::MAIN: Output directory set to: X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 DEBUG   ::MAIN: Scanning files in directory: X:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 DEBUG   ::MAIN: Found 1 files in X:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 INFO    ::COPYLINK: MEDIAFILE: [Test.Video.S01E02.720p.WEB.x264-TEST.mkv]
2018-06-29 12:59:13 INFO    ::COPYLINK: SOURCE FOLDER: [X:\Download\tv]
2018-06-29 12:59:13 INFO    ::COPYLINK: TARGET FOLDER: [X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv]
2018-06-29 12:59:13 INFO    ::COPYLINK: Hard linking SOURCE MEDIAFILE -> TARGET FOLDER
2018-06-29 12:59:13 DEBUG   ::MAIN: Checking for archives to extract in directory: X:\Download\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 INFO    ::MAIN: FLATTEN: Flattening directory: X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 DEBUG   ::MAIN: Checking for empty folders in:X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 INFO    ::MAIN: Found 1 media files in X:\Process\tv\Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 INFO    ::MAIN: Calling SickBeard:tv to post-process:Test.Video.S01E02.720p.WEB.x264-TEST.mkv
2018-06-29 12:59:13 DEBUG   ::SERVER: Attempting to connect to server at http://localhost:40030
2018-06-29 12:59:13 DEBUG   ::SERVER: Server responded at http://localhost:40030
2018-06-29 12:59:13 INFO    ::MAIN: SickBeard:tv fork set to SickGear
2018-06-29 12:59:13 POSTPROCESS::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
2018-06-29 12:59:13 DEBUG   ::SICKBEARD: Opening URL: http://localhost:40030/home/postprocess/processEpisode with params: {'nzbName': 'Test.Video.S01E02.720p.WEB.x264-TEST.mkv', 'quiet': 1, 'failed': 0, 'dir': 'X:\\Process\\tv\\Test.Video.S01E02.720p.WEB.x264-TEST.mkv'}
2018-06-29 12:59:13 ERROR   ::SICKBEARD: Server returned status 401
2018-06-29 12:59:13 ERROR   ::MAIN: A problem was reported in the autoProcess* script. If torrent was paused we will resume seeding
2018-06-29 12:59:13 DEBUG   ::MAIN: Starting torrent Test.Video.S01E02.720p.WEB.x264-TEST.mkv in utorrent
2018-06-29 12:59:18 ERROR   ::MAIN: A problem was reported in the C:\nzbToMedia\TorrentToMedia.py script.
```
Did a quick check to see if if the http://localhost:40030/ was accessible and it just gave me a prompt so that shouldn't be the issue.

Sickgear is at 

`BRANCH: master / COMMIT: 54a877d3e39ad7b33d7bebedfa5b91c4b2e05242`

I'll try to rollback and see what happens.
